src/cli/tc.rs

```diff
@@ -111,3 +114,19 @@ fn display_config(config: &Config, n_plus_one_queries: bool) {
   let seq = vec![Fmt::n_plus_one_data(n_plus_one_queries, config)];
   Fmt::display(Fmt::table(seq));
 }
+
+fn write_output_file(filename: &String, config: &Config) -> Result<()> {
+  let filetype = Source::detect(filename)?;
+
+  let contents = match filetype {
```

Comment:
checkout `source.rs` it has a lot of helper methods that can simplify this code.

Reply:
Hi @tusharmath, I have already used the Detect method from there. I could not see any other methods that could be used from there. Please let me know if there is something you see that could be re-used. Thanks.